dolphinscheduler-extract/dolphinscheduler-extract-common/src/main/java/org/apache/dolphinscheduler/extract/common/service/impl/LogServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+
+import org.apache.dolphinscheduler.common.utils.FileUtils;
+import org.apache.dolphinscheduler.common.utils.LogUtils;
+import org.apache.dolphinscheduler.extract.common.ILogService;
+import org.apache.dolphinscheduler.extract.common.transportor.TaskInstanceLogFileDownloadRequest;
+import org.apache.dolphinscheduler.extract.common.transportor.TaskInstanceLogFileDownloadResponse;
+import org.apache.dolphinscheduler.extract.common.transportor.TaskInstanceLogPageQueryRequest;
+import org.apache.dolphinscheduler.extract.common.transportor.TaskInstanceLogPageQueryResponse;
+
+import java.util.List;
+
+public class LogServiceImpl implements ILogService {
+
+    /**
+     * Downloads the entire log file for a task instance.
+     *
+     * @param taskInstanceLogFileDownloadRequest Request object containing the path to the task instance log file.
+     * @return Response object containing the log file content in byte array form.
+     */
+    @Override
+    public TaskInstanceLogFileDownloadResponse getTaskInstanceWholeLogFileBytes(TaskInstanceLogFileDownloadRequest taskInstanceLogFileDownloadRequest) {
+        final TaskInstanceLogFileDownloadResponse taskInstanceLogFileDownloadResponse =
+                new TaskInstanceLogFileDownloadResponse();
+        try {
+            byte[] bytes = LogUtils
+                    .getFileContentBytesFromLocal(taskInstanceLogFileDownloadRequest.getTaskInstanceLogAbsolutePath());
+            taskInstanceLogFileDownloadResponse.setLogBytes(bytes);
+        } catch (Exception e) {
+            taskInstanceLogFileDownloadResponse.setCode(1);
+            taskInstanceLogFileDownloadResponse.setMessage(e.getMessage());
+        }
+        return taskInstanceLogFileDownloadResponse;
+    }
+
+    /**
+     * Performs paginated queries on task instance logs.
+     *
+     * @param taskInstanceLogPageQueryRequest Request object containing the path to the task instance log file, the number of lines to skip, and the maximum number of lines to read.
+     * @return Response object containing the log content.
+     */
+    @Override
+    public TaskInstanceLogPageQueryResponse pageQueryTaskInstanceLog(TaskInstanceLogPageQueryRequest taskInstanceLogPageQueryRequest) {
+        final TaskInstanceLogPageQueryResponse taskInstanceLogPageQueryResponse =
+                new TaskInstanceLogPageQueryResponse();
+        List<String> lines;
+        try {
+            lines = LogUtils.readPartFileContentFromLocal(
+                    taskInstanceLogPageQueryRequest.getTaskInstanceLogAbsolutePath(),
+                    taskInstanceLogPageQueryRequest.getSkipLineNum(),
+                    taskInstanceLogPageQueryRequest.getLimit());
+            taskInstanceLogPageQueryResponse.setLogContent(LogUtils.rollViewLogLines(lines));
+        } catch (Exception e) {
+            taskInstanceLogPageQueryResponse.setCode(1);
+            taskInstanceLogPageQueryResponse.setMessage(e.getMessage());
```

Comment:
Use ExceptionUtils to get the root cause message.

Reply:
change to   ExceptionUtils.getMessage